src/layouts/components/HeaderNetworkMenu.tsx

```diff
@@ -0,0 +1,121 @@
+import { useNetwork } from "@contexts/NetworkContext";
+import { getEnvironment } from "@contexts/Environment";
+import { Menu, Transition } from "@headlessui/react";
+import { CgChevronDown, CgChevronUp } from "react-icons/cg";
+import { MdCheckCircle } from "react-icons/md";
+import React, { Fragment, useState } from "react";
+import { NetworkIcon } from "@components/icons/NetworkIcon";
+import clsx from "clsx";
+
+export function HeaderNetworkMenu(): JSX.Element {
+  const connection = useNetwork().connection;
+  const { networks } = getEnvironment();
+
+  return (
+    <div className="flex">
+      <Menu as="div" className="w-full flex relative">
+      {({ open }) => (
+        <>
+          <Menu.Button className="flex items-center w-full justify-between">
+            <div className="flex items-center mr-3.5">
+              <NetworkIcon className="fill-[#00AD1D]" />
+              <div className="text-xl ml-2 leading-none text-white-50">
+                {connection}
+              </div>
+            </div>
+            {open
+              ? <CgChevronUp className="h-6 w-6 text-white-700" />
+              : <CgChevronDown className="h-6 w-6 text-white-700" />
```

Comment:
can work on it at separate PR too: can we transform the button to 180 degree on click so we can have a transition effect?